From a3ace31149b633f3c1f647d9ec06d622db2d4aa1 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 14 Dec 2013 13:18:48 +0000
Subject: [PATCH] builder: Move is_block_device function into common utilities.

This is just code motion.

(cherry picked from commit 81b4e5d5cd68f65aa950251e22332b3fa477fecb)
---
 builder/builder.ml    | 3 ---
 mllib/common_utils.ml | 4 ++++
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/builder/builder.ml b/builder/builder.ml
index 2710f10..b59f6ec 100644
--- a/builder/builder.ml
+++ b/builder/builder.ml
@@ -249,9 +249,6 @@ let main () =
    * - did the user specify --size?
    *)
   let output, size, format, delete_output_file, do_resize, resize_sparse =
-    let is_block_device file =
-      try (stat file).st_kind = S_BLK
-      with Unix_error _ -> false
     in
 
     let headroom = 256L *^ 1024L *^ 1024L in
diff --git a/mllib/common_utils.ml b/mllib/common_utils.ml
index b2ea783..93d22fd 100644
--- a/mllib/common_utils.ml
+++ b/mllib/common_utils.ml
@@ -391,6 +391,10 @@ let rm_rf_only_files (g : Guestfs.guestfs) dir =
     List.iter g#rm files
   )
 
+let is_block_device file =
+  try (Unix.stat file).Unix.st_kind = Unix.S_BLK
+  with Unix.Unix_error _ -> false
+
 let is_char_device file =
   try (Unix.stat file).Unix.st_kind = Unix.S_CHR
   with Unix.Unix_error _ -> false
-- 
1.8.3.1

